File: 01findthenumber/solution.py
import sys

low = 1
high = 2 * 10**9

while True:
    n = int((low + high) / 2)
    print(n)
    sys.stdout.flush()

    RESULT = str(sys.stdin.readline()).strip()
    if RESULT == 'down':
        high = n
    elif RESULT == 'up':
        low = n
    else:
        break

File: 01findthenumber/solution.js
const low = 1;
const high = 2 * (10 ** 9)

while (true) {
    const n = int((low + high) / 2)
    print(n)

    const result = gets().trim();
    if (result == 'FLOATS') {
        high = n;
    } else {
        low = n;
    }
}

''.cha